firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -1000,10 +1011,22 @@ class BrowserViewController: UIViewController,
         )
         addressToolbarContainer.applyTheme(theme: currentTheme())
         addressToolbarContainer.addToParent(parent: isBottomSearchBar ? overKeyboardContainer : header)
+
+        guard isSwipingTabsEnabled else { return }
+        addressBarPanGestureHandler = AddressBarPanGestureHandler(
+            contentContainer: contentContainer,
+            addressBarContainer: isBottomSearchBar ? overKeyboardContainer : header,
+            webPagePreview: webPagePreview,
+            tabManager: tabManager,
+            windowUUID: windowUUID,
+            screenshotHelper: screenshotHelper
+        )
+        webPagePreview.updateLayoutBasedOn(searchBarPosition: searchBarPosition)
```

Comment:
this seems like it makes sense as state that changes when the layout is updated. Curious to see if we thought of using Redux for this and why we ultimately decided not to use that pattern in case I'm missing something

Reply:
Honestly, haven't thought about using Redux pattern at all in this case, but I am willing to. Do you think I should update the implementation and use Redux?

Reply:
Thanks for making the updates in the Redux pattern. Looking at the new changes, it seems we created a state based on `addressBarPosition`. Could we use the bar position from the toolbar state instead as the single source of truth? 

There are some areas where I'm unclear such as while we need `addressBarPanGestureHandler` on the `BVC` as opposed to decoupling, might be helpful to chat live if that works for you too!